rust/index/bindings.cpp

```diff
@@ -94,6 +94,7 @@ class Index
             throw std::runtime_error("Index already inited");
         }
         appr_alg = new hnswlib::HierarchicalNSW<dist_t>(l2space, path_to_index, false, 0, allow_replace_deleted, normalize, is_persistent_index);
+        appr_alg->checkIntegrity();
```

Comment:
Can we add a rust level test that somehow corrupts the index and verifies we get the error?

Reply:
I've added a rust unit test that corrupts the index and verifies the error message.